### PR TITLE
[Merged by Bors] - chore: cache cancelled started builds

### PR DESCRIPTION
**Pull request by kim-em**
Per https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Some.20files.20not.20found.20in.20the.20cache/near/439945625

### CODE REVIEW

File: .github/workflows/bors.yml [hunk 1]
```diff
@@ -205,9 +206,10 @@ jobs:
             bash -o pipefail -c "env LEAN_ABORT_ON_PANIC=1 lake build --wfail -KCI"
 
       - name: upload cache
-        # We only upload the cache if the build ran (either succeeding or failing),
-        # but not if it was skipped.
-        if: ${{ always() && steps.build.outcome == 'success' || steps.build.outcome == 'failure' }}
+        # We only upload the cache if the build started (whether succeeding, failing, or cancelled)
+        # but not if any earlier step failed.
+        # See discussion at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Some.20files.20not.20found.20in.20the.20cache/near/407183836
+        if: ${{ always() && steps.get.outcome == 'success' }}
```
Comment (adomani) [inline .github/workflows/bors.yml:212]: Shouldn't this also mention `cancelled`?  Or maybe the build CI step should be labelled as `continue-on-error`?

File: .github/workflows/bors.yml [hunk 2]
```diff
@@ -205,9 +206,10 @@ jobs:
             bash -o pipefail -c "env LEAN_ABORT_ON_PANIC=1 lake build --wfail -KCI"
 
       - name: upload cache
-        # We only upload the cache if the build ran (either succeeding or failing),
-        # but not if it was skipped.
-        if: ${{ always() && steps.build.outcome == 'success' || steps.build.outcome == 'failure' }}
+        # We only upload the cache if the build started (whether succeeding, failing, or cancelled)
+        # but not if any earlier step failed.
```
Comment (semorrison) [inline .github/workflows/bors.yml:210]:
```suggestion
        # but not if any earlier step failed or was cancelled.
```

Comment (semorrison) [reply .github/workflows/bors.yml:210]: @adomani, is that what you meant?

Comment (adomani) [reply .github/workflows/bors.yml:210]: This was not what I had in mind, but it certainly is good to mention it!

I think that I was confused and in my mind I was still thinking that you were splitting on the `build` step, not on the `get` step!

**Comment by adomani** [comment]
Thanks!  Up to modifying the comment, I am happy to

maintainer merge

this PR!

**Comment by jcommelin** [comment]
bors d+

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**